p2p/src/channels/rpc/p2p_channels_rpc_reducer.rs

```diff
@@ -238,10 +248,10 @@ impl P2pChannelsRpcState {
                     return Ok(());
                 }
 
-                dispatcher.push(P2pChannelsRpcEffectfulAction::ResponseSend {
+                dispatcher.push(P2pChannelsEffectfulAction::RequestSend {
                     peer_id,
-                    id,
-                    response,
+                    msg_id: MsgId::first(),
+                    msg: ChannelMsg::Rpc(RpcChannelMsg::Response(id, response.map(|v| *v))),
```

Comment:
Ok, I see that now even "response" messages are wrapped into a `RequestSend` variant. I'm not fully convinced about this approach.

Reply:
I could make one action `RequestSend` and one `ResponseSend` to make it clearer, or change name to something like `MessageSend`

Reply:
@0xMimir please spend more time explaining anything relevant about the changes in the summary of the issue so that reviewers don't need to figure it on their own. It has to be very clear what is going on and why this is a worthwhile improvement. There is obviously more going on than what is explained in the summary.